client/src/components/CreatePost.tsx

```diff
@@ -1,181 +1,281 @@
-import React, { useState } from "react";
+import React, { useState } from 'react';
 
-import Grid from "@mui/material/Grid";
-import Box from "@mui/material/Box";
-import Container from "@mui/material/Container";
-import Button from "@mui/material/Button";
-import TextField from "@mui/material/TextField";
-import Divider from "@mui/material/Divider";
-import Typography from "@mui/material/Typography";
-import Tooltip from "@mui/material/Tooltip";
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
+import Container from '@mui/material/Container';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import Divider from '@mui/material/Divider';
+import Typography from '@mui/material/Typography';
+import Tooltip from '@mui/material/Tooltip';
+import Dialog from '@mui/material/Dialog';
+import ArrowBack from '@mui/icons-material/ArrowBack';
+import PreviewPopUp from './PreviewPopUp';
+import Snackbar from '@mui/material/Snackbar';
 
-import PreviewPopUp from "./PreviewPopUp";
+import ServerApi from '../api/v1/index';
+
+const api = new ServerApi();
 
 function CreatePost() {
-  // create hooks for input fields
+  // hooks
   const [form, setForm] = useState({
-    title: "",
-    body: "",
-    file: "",
-    tags: "",
-    eventCapacity: "",
-    eventLocation: "",
+    title: '',
+    body: '',
+    file: '',
+    tags: '',
+    capacity: '',
```

Comment:
Shouldn't capacity be an integer? Default can be -1 or something for an unset capacity.